web/src/components/navbar/NavBarDashboardLink.test.tsx

```diff
@@ -0,0 +1,95 @@
+import { NavBarDashboardLink } from "@/components/navbar/NavBarDashboardLink";
+import { ROUTES } from "@/lib/domain-logic/routes";
+import analytics from "@/lib/utils/analytics";
+import { mockPush, useMockRouter } from "@/test/mock/mockRouter";
+import {
+  WithStatefulUserData,
+  currentUserData,
+  setupStatefulUserDataContext,
+} from "@/test/mock/withStatefulUserData";
+import { Business } from "@businessnjgovnavigator/shared";
+import {
+  generateBusiness,
+  generateUser,
+  generateUserDataForBusiness,
+} from "@businessnjgovnavigator/shared/test";
+import { UserData, createEmptyUserData } from "@businessnjgovnavigator/shared/userData";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+
+jest.mock("next/router", () => ({ useRouter: jest.fn() }));
+jest.mock("@/lib/utils/analytics", () => setupMockAnalytics());
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+
+function setupMockAnalytics(): typeof analytics {
+  return {
+    ...jest.requireActual("@/lib/utils/analytics").default,
+    event: {
+      ...jest.requireActual("@/lib/utils/analytics").default.event,
+      my_account: {
+        click: {
+          my_account: jest.fn(),
+        },
+      },
+    },
+  };
+}
+const mockAnalytics = analytics as jest.Mocked<typeof analytics>;
+
+const renderComponent = ({
+  userData,
+  previousBusiness,
+}: {
+  userData?: UserData | undefined;
+  previousBusiness?: Business | undefined;
+}): void => {
+  render(
+    <WithStatefulUserData
```

Comment:
dust - why do we need `WithStatefulUserData`?

Reply:
this is needed for the third test to check to see if we properly update the userData in order to return to the previous business